app/controllers/dashboard_controller.rb

```diff
@@ -14,6 +14,16 @@ def index
     end
   end
 
+  def show
+    if @widget.present? && @widget.user == User.current
+      render(:partial => @widget.template, :locals => @widget.data)
```

Comment:
its logged only when it finished rendering,  i had of a problem to figure
out which widget was rendering, but i guess i could go both ways.